apps/vendor/models.py

```diff
@@ -0,0 +1,161 @@
+from django.db import models
+from apps.customer.models import LedgerAccounts
+from apps.masters.models import FirmStatuses, GstCategories, PriceCategories, Territory, Transporters
+from utils_variables import vendorcategory, vendorpaymentterms, vendoragent, vendor, vendorattachments, vendoraddresses
+from utils_methods import custom_upload_to
+
+# Create your models here.
+    
+class VendorCategory(models.Model):
+    vendor_category_id = models.AutoField(primary_key=True)
+    code = models.CharField(max_length=50, null=True, default=None)
+    name = models.CharField(max_length=255, null=True, default=None)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.vendor_category_id}_{self.name} ({self.code})'
+
+    class Meta:
+        db_table = vendorcategory
+
+
+class VendorPaymentTerms(models.Model):
+    payment_term_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=255, null=True, default=None)
+    code = models.CharField(max_length=50, null=True, default=None)
+    fixed_days = models.IntegerField(null=True, default=None)
+    no_of_fixed_days = models.IntegerField(null=True, default=None)
+    payment_cycle = models.CharField(max_length=255, null=True, default=None)
+    run_on = models.CharField(max_length=255, null=True, default=None)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f'{self.payment_term_id}_{self.name} ({self.code})'
+    
+    class Meta:
+        db_table = vendorpaymentterms
+
+
+class VendorAgent(models.Model):
+    vendor_agent_id = models.AutoField(primary_key=True)
+    code = models.CharField(max_length=50, null=True, default=None)
+    name = models.CharField(max_length=255, null=True, default=None)
+    commission_rate = models.DecimalField(max_digits=18, decimal_places=2, null=True, default=None)
+    
+    RATE_ON_CHOICES = [
+        ('Qty', 'Quantity'),
+        ('Amount', 'Amount'),
```

Comment:
This should have another value 'Both' Can you double check

Reply:
Following is the SQL structure Thulasi, there is no 'both' option.
    rate_on ENUM("Qty", "Amount"),
    amount_type ENUM("Taxable", "BillAmount"),